[#9251] Remove/Replace methods *WithoutDocument in StudentsDb

priyamalM opened the pull request:
Fixes #9251

remove * withoutDocument methods in StudentDb class and removed their usages,
couldn't remove the boolean parameter requested in the methods which replaced withoutDocument
methods as there were several scenarios where values true and false both were used.

priyamalM commented:
PR is ready for review, please go through them and I'll be sure to change all the requested changes if there is any.  👍

xpdavid commented on src/main/java/teammates/logic/core/StudentsLogic.java:
```diff
@@ -462,15 +462,15 @@ public void deleteStudentsForGoogleId(String googleId) {
+        studentsDb.deleteStudentsForGoogleId(googleId, true);
     }
 
     public void deleteStudentsForGoogleIdWithoutDocument(String googleId) {
```
`deleteStudentsForGoogleIdWithoutDocument` still exist?

xpdavid commented on src/main/java/teammates/storage/api/StudentsDb.java:
```diff
@@ -336,14 +331,6 @@ private void updateStudentDetails(String newName, String newTeamName, String new
-        deleteStudent(courseId, email, false);
-    }
-
     public void deleteStudent(String courseId, String email, boolean hasDocument) {
```
The parameter `boolean hasDocument` should be removed instead.

I think it is OK to delete document even though it might not exist (although it should exist for each student).

priyamalM replied:
removed the boolean parameter from all the methods which delete students. since document exists for each student, can I remove the boolean parameter from `create & update` student methods as well.

xpdavid replied:
Yes. That is what the issue about.

priyamalM replied:
I removed hasDocument param from creating and updating methods as well.

xpdavid commented on src/main/java/teammates/storage/api/StudentsDb.java:
```diff
@@ -392,14 +371,6 @@ public void deleteStudentsForGoogleId(String googleId, boolean hasDocument) {
-        deleteStudentsForCourse(courseId, false);
-    }
-
     public void deleteStudentsForCourse(String courseId, boolean hasDocument) {
```
Same here.

priyamalM commented:
I removed hasDocument parameter from creating and updating a student and changed test cases with respect to that

wkurniawan07 commented:
@priyamalM also check and remove methods ending with `*WithoutSearchability` as well.

After this fix, searching the following keywords in the code base should result in a miss:
- `hasDocument`
- `withDocument`
- `withoutDocument`
- `withSearchability`
- `withoutSearchability`

priyamalM commented:
> @priyamalM also check and remove methods ending with `*WithoutSearchability` as well.
> 
> After this fix, searching the following keywords in the code base should result in a miss:
> 
> * `hasDocument`
> * `withDocument`
> * `withoutDocument`
> * `withSearchability`
> * `withoutSearchability`

thanks for the reply, I'll start working on this

priyamalM commented:
**PR** is ready for review 😃 I removed all the occurrences @wkurniawan07 mentioned and I removed the boolean parameter `keepUpdateTimestamp` from the `updatingStudent` method which I thought is not needed since it is always **false**, except for one test case.

wkurniawan07 commented:
> I removed the boolean parameter `keepUpdateTimestamp` from the `updatingStudent` method which I thought is not needed since it is always **false**

Let's not mix two issues in one PR. This is what #9250 is about.

priyamalM commented:
> > I removed the boolean parameter `keepUpdateTimestamp` from the `updatingStudent` method which I thought is not needed since it is always **false**
> 
> Let's not mix two issues in one PR. This is what #9250 is about.

I revert the commit back and kept the `KeepUpdateTimestamp` since it refers to another issue. PR is ready for review

xpdavid commented on src/test/java/teammates/test/cases/search/StudentSearchTest.java:
```diff
@@ -88,14 +88,14 @@ public void allTests() {
         ______TS("success: search for students; deleted student without deleted document: the document "
```
You should modify/remove the test case as fit

priyamalM replied:
I removed the test case since deletion is already checked

xpdavid commented on src/test/java/teammates/test/cases/logic/StudentsLogicTest.java:
```diff
@@ -69,7 +69,7 @@ public void testAll() throws Exception {
 
         testValidateSections();
-        testupdateStudentCascadeWithoutDocument();
+        testupdateStudentCascade();
```
Camel case naming -> `testUpdateStudentCascade`

priyamalM replied:
i didn't notice this since it was already not camel case . bt i changed it now.

priyamalM commented on src/test/java/teammates/test/cases/logic/StudentsLogicTest.java:
```diff
@@ -590,7 +590,7 @@ private void testEnrollLinesChecking() throws Exception {
      * Returns the error message of EnrollException thrown when trying to call
      * {@link StudentsLogic#createStudents(String, String)} method with
      * {@code invalidEnrollLines}. This method assumes that an EnrollException is thrown, else this method fails with
-     * {@link #signalFailureToDetectException(String...)} ()}.
```
this was giving an error in my IDE that's why I removed it, I shouldn't have deleted it cuz it's not related.

priyamalM commented:
@xpdavid  PR is ready for review, I resolved the conflict with respect to this [#9250] as well. (https://github.com/TEAMMATES/teammates/issues/9250)
